### PR TITLE
Add JOSS paper to repo

### LINKED ISSUE #257: prepare JOSS submission
https://joss.readthedocs.io/en/latest/submitting.html#

https://joss.theoj.org/papers/new 

Draft md file: https://hackmd.io/ELc-ybBpTMeKQ34CPhYTPA?edit

### PR DESCRIPTION
**Pull request by rwood-97**
### Summary

This PR adds the JOSS paper and references file to the repo in a directory called `paper`.
It also adds a GH action which builds a draft of the paper - we should turn this off once we are done making changes!

Fixes #257 

## To do still

- [x] Check output of GH action
- [x] Fill in title, AHRC grant no., anything else that is currently "XXXX".

### CODE REVIEW

File: paper/paper.md
Comment (kasra-hosseini) [inline paper/paper.md:65]: @kmcdono2 maybe ref for `Since the 1990s, map libraries have been scanning maps and creating digital collections of these images.`?

Comment (kasra-hosseini) [reply paper/paper.md:65]: @kmcdono2 also a ref for `more than a million images` would be helpful

File: paper/paper.md
Comment (kasra-hosseini) [inline paper/paper.md:82]: CV / Computer Vision / Curriculum Vitae ... maybe we can choose another title here?

File: paper/paper.md
Comment (kasra-hosseini) [inline paper/paper.md:95]: Should we add a sentence or two about documentation/tutorials/release here?

**Comment by kasra-hosseini** [comment]
@rwood-97 @kmcdono2 @dcsw2 Thank you for preparing this 🙏 . I made some minor changes + add some inline comments / questions. Otherwise, it is good to go from my side.

File: paper/paper.md
Comment (kmcdono2) [reply paper/paper.md:95]: We have a word limit, but I'll try!

File: paper/paper.md
Comment (kmcdono2) [reply paper/paper.md:82]: Good point.

File: paper/paper.md
Comment (kmcdono2) [reply paper/paper.md:65]: Yes, we can add a ref for the first, but for the second this is based on my count. I'm writing about this elsewhere, but not yet published.

**Review by kmcdono2 — CHANGES_REQUESTED** [review]
I've addressed @kasra-hosseini's comments and I think this is good to go once those new pull requests are merged.

File: paper/paper.md
Comment (kmcdono2) [inline paper/paper.md:82]: Change to "Project Curriculum Vitae"

**Comment by rwood-97** [comment]
@kmcdono2 Think this is good to go, version is 1.1.0 now though AHRC grant no. is still empty (as discussed)

**Review by kmcdono2 — APPROVED** [review]

**Pull request merged**
